### ISSUE TITLE
ENH : Automatic ROM reporting

### ISSUE THREAD

**Issue by KikeM** [KikeM/msc-thesis#35]
Once the ROM is defined in terms of the weak formulations (mass, stiffness, forcings, etc.),  boundary conditions, parameter space, and hyper reductions, it would be good to automate the "reporting".

Questions to answer in a ROM reporting:
- Which problem is solved.
- Efficiency rates: how much was the problem compressed.
- Errors: L2, H1.
  - Time integration errors. 
  - Take a look at the paper from Negri, he suggested a couple of metrics. 
  - A posteriori errors.
- Error location: is it near a non-linearity?
- Spectral decay in the POD/SVD.
- How far away can we go from the trained set? (This actually proves if we have captured the dynamics or not).
  - Euclidean distance from the furthest point of the training set.
  - Euclidean distance from the "centre" of the cluster.  
- Performance rates: how much time is saved up (this metric is implementation-dependent, be careful).

**Comment by KikeM** [comment]
I think I am coming up with a sufficiently flexible code structure, so perhaps it is a good time to wrap up the ROM integrator. 

That is, in the same way that I have `romtime.base.OneDimensionalSolver`, which I inherit in `romtime.heat.HeatEquationSolver` to define the specific implementation of the heat equation; perhaps I should have a sort of `romtime.problems.heat.HeatEquationRom`, where I would simply generalise a bit the test `tests/test_mfp1.py::test_rom_deim_mdeim`.
